### PR TITLE
[TG-59] JoinGame and GameSessionsRepository implemented

### PR DESCRIPTION
**Pull request by NickNovic**

### CODE REVIEW

File: Source/TileGameServer/TileGameServer.DataAccess/Entities/GameSession.cs [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+using System.Collections.Generic;
+using TileGameServer.Infrastructure.Models;
 
 namespace TileGameServer.DataAccess.Entities
 {
     public class GameSession : BaseEntity
     {
         public GameSessionStatus Status { get; set; }
         public DateTime CreationDate { get; set; }
+        public List<Guid> PlayersId { get; set; } = new List<Guid>();
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer.DataAccess/Entities/GameSession.cs:13]: PlayerIds

File: Source/TileGameServer/TileGameServer.DataAccess/Repositories/ListGameSessionsRepository.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            => await Task.Run(() => 
+            GameSessions.Exists(t 
+            => t.PlayersId.FirstOrDefault(a => a == playerId) != null));
+        
+
+        public async Task<GameSession> GetAsync(Guid id) => await Task.Run(() 
+            => GameSessions.FirstOrDefault(t => t.Id == id));
+        
+
+        public Task SaveChangesAsync()
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer.DataAccess/Repositories/ListGameSessionsRepository.cs:36]: Вынести в новый интерфейс

File: Source/TileGameServer/TileGameServer/Commands/Menu/JoinGameSession.cs [hunk 3]
```diff
@@ -13,16 +15,31 @@ public class JoinGameSession
...
         }
 
         public class JoinGameSessionCommandHandler : IRequestHandler<JoinGameSessionCommand, JoinGameSessionResponse>
         {
-            public Task<JoinGameSessionResponse> Handle(JoinGameSessionCommand request, CancellationToken cancellationToken)
+            private IListGameSessionsRepository ListGameSessionsRepository { get; }
+            public async Task<JoinGameSessionResponse> Handle(JoinGameSessionCommand request, CancellationToken cancellationToken)
             {
-                return Task.FromResult(new JoinGameSessionResponse
+                if(!await ListGameSessionsRepository.ExistsWithPlayerAsync(request.UserId))
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer/Commands/Menu/JoinGameSession.cs:26]: If(exists)
{
return
}

File: Source/TileGameServer/TileGameServer/Controllers/MenuController.cs [hunk 4]
```diff
@@ -29,11 +29,12 @@ public async Task<ActionResult<Unit>> CreateGame()
         }
 
         [HttpGet("joinGame")]
-        public async Task<ActionResult<Unit>> JoinGame()
+        public async Task<ActionResult<Unit>> JoinGame([FromBody] Guid SessionId)
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer/Controllers/MenuController.cs:32]: Заменить на DTO

File: Source/TileGameServer/TileGameServer/Controllers/MenuController.cs [hunk 5]
```diff
@@ -29,11 +29,12 @@ public async Task<ActionResult<Unit>> CreateGame()
...
         {
             var command = new JoinGameSession.JoinGameSessionCommand
             {
-                UserId = Guid.Empty
+                UserId = Guid.Empty,
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer/Controllers/MenuController.cs:36]: Использовать User.GetClaim(AppClaimTypes.UserId);

File: Source/TileGameServer/TileGameServer.DataAccess/Repositories/ListGameSessionsRepository.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Threading.Tasks;
+using TileGameServer.DataAccess.Entities;
+using System.Collections.Generic;
+using System.Linq;
+using TileGameServer.Infrastructure.Models;
+
+namespace TileGameServer.DataAccess.Repositories
+{
+    public class ListGameSessionsRepository : IListGameSessionsRepository
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer.DataAccess/Repositories/ListGameSessionsRepository.cs:10]: Переименовать в GameSessionRepository

File: Source/TileGameServer/TileGameServer.DataAccess/Repositories/IListGameSessionsRepository.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Threading.Tasks;
+using TileGameServer.DataAccess.Entities;
+using TileGameServer.DataAccess.Enums;
+using TileGameServer.DataAccess.Repositories.Generic;
+using TileGameServer.Infrastructure.Models;
+
+namespace TileGameServer.DataAccess.Repositories
+{
+    public interface IListGameSessionsRepository : IRepository<GameSession>
```
Comment (ArtiomCiobanu) [inline Source/TileGameServer/TileGameServer.DataAccess/Repositories/IListGameSessionsRepository.cs:10]: Перенести метод в IGameSessionRepository

Comment (ArtiomCiobanu) [reply Source/TileGameServer/TileGameServer.DataAccess/Repositories/IListGameSessionsRepository.cs:10]: И удалить этот интерфейс

**Review by Cute-Monster — CHANGES_REQUESTED** [review]
1. Make a new branch from current according to the new structure.
2. Fix the mentioned issues

**Comment by Cute-Monster** [comment]
@NickNovic, I'm closing this PR, cause ths actions shoul be performed

>
>     1. Make a new branch from current according to the new structure.
> 
>     2. Fix the mentioned issues

**Pull request closed without merging**